lib/bot.js

```diff
@@ -91,6 +91,15 @@ class Bot {
       ...this.ircOptions
     };
 
+    // default encoding to UTF-8 so messages to Discord aren't corrupted
+    if (!('encoding' in ircOptions)) {
```

Comment:
`if (!ircOptions.encoding)` or if you really want to explicitly check that the property isn't set: `!ircOptions.hasOwnProperty('encoding')`

Reply:
The linter seems to complain about `ircOptions.hasOwnProperty('encoding')`:

```
.../discord-irc/lib/bot.js
  95:21  error  Do not access Object.prototype method 'hasOwnProperty' from target object  no-prototype-builtins
```

I've made it use `Object.prototype.hasOwnProperty.call(ircOptions, 'encoding')` instead.

I think I decided to check for the property because if `encoding` is set without support for translating encodings, it will output errors for each message it attempts (and fails) to convert the encoding of. So, if the library erroneously says it can convert an encoding, and so `discord-irc` sets the encoding, (or if something else goes wrong along the way) it could be useful for people to manually disable the option by setting it to null.

Reply:
Also – what's the difference between `'encoding' in ircOptions` and `ircOptions.hasOwnProperty('encoding')`? Whether the object `ircOptions` implements it itself, versus whether some superclass of it does?

Reply:
Yep: https://stackoverflow.com/questions/13632999/if-key-in-object-or-ifobject-hasownpropertykey

Probably doesn't really matter here though, but I think in general `hasOwnProperty` is considered a bit more hygienic.